packages/core/src/ToastProvider/ToastProvider.spec.tsx

```diff
@@ -0,0 +1,77 @@
+// @ts-nocheck
+jest.useFakeTimers()
+
+jest.mock('react-dom', () => ({
+  ...jest.requireActual('react-dom'),
+  createPortal: (children) => children,
+}))
+
+import React from 'react'
+import { act, fireEvent, render, screen } from '../__test__/testing-library'
+import { ToastProvider } from '..'
+import { MockToastChildren } from '../__test__/mocks/toasts'
+
+describe('ToastProvider', () => {
+  it('renders success toast and clears it', async () => {
+    render(
+      <ToastProvider>
+        <MockToastChildren />
+      </ToastProvider>
+    )
+
+    expect(screen.queryByText('Success Toast Message')).not.toBeInTheDocument()
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    expect(screen.queryByText('Success Toast Message')).toBeInTheDocument()
+
+    const closeIcon = screen.getAllByTitle('close-toast')[0]
+
+    act(() => {
+      fireEvent.click(closeIcon)
+      jest.runAllTimers()
+    })
+
+    expect(screen.queryByText('Success Toast Message')).not.toBeInTheDocument()
+  })
+
+  it('renders the max number of toasts', () => {
+    render(
+      <ToastProvider domRootId='test' maxToasts={4}>
+        <MockToastChildren />
+      </ToastProvider>
+    )
+
+    expect(screen.queryByText('Success Toast')).not.toBeInTheDocument()
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    fireEvent.click(screen.queryByText('Add Success Toast'))
+    expect(screen.queryAllByText('Success Toast Message').length).toBe(4)
+
+    act(() => {
+      fireEvent.click(screen.getAllByTitle('close-toast')[0])
+      jest.runAllTimers()
+    })
+
+    expect(screen.queryAllByText('Success Toast Message').length).toBe(4)
+
+    act(() => {
+      fireEvent.click(screen.getAllByTitle('close-toast')[0])
+      jest.runAllTimers()
+    })
+
+    expect(screen.queryAllByText('Success Toast Message').length).toBe(3)
+  })
```

Comment:
I'm not really sure I get this part 🤔 you close the toast but still have 4?

Reply:
maxToasts is set to 4. I create 5, 4 are shown. I remove 1, there are still 4 being shown. Then I remove another 1 and you get 3 left. This test is making sure that the maxToasts prop works correctly

Reply:
oh gotcha. you have 5 toasts but only rendering 4 because maxToast is 4.